src/intersections/lineraysegmentplane.jl

```diff
@@ -0,0 +1,111 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function lineraysegmentdirection(x::Line)
+  return x(1) - x(0)
+end
+
+function lineraysegmentdirection(x::Ray)
+  return direction(x)
+end
+
+function lineraysegmentdirection(x::Segment)
+  return -reduce(-, coordinates.(vertices(x)))
+end
```

Comment:
Same here, can all these three methods be replaced by the single generic x(1) - x(0)?

Reply:
Totally can be replaced with an x(1) - x(0), but I think it introduces unnecessary operations? Ray already contains the direction information and for segment subtracting requires less operations than calculating `a + t * (b - a)` for t = 0 and t = 1. In my application (and I'm sure others too) the runtime is already quite long and it seems to me making these lower level functions simpler will help :)

Reply:
Isn't this a case of premature optimization? If we just make sure that the code is general and works for the three types first, we can come later to benchmark and spot the bottlenecks. My guess is that this tiny detail won't be the bottleneck of most applications. Let's make the code work first before we start with micro-optimizations 👍🏽